test/functional/diagnose/diagnose-also-add.bats

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env bats
+
+# Author: Otavio Pontes
+# Email: otavio.pontes@intel.com
+
+load "../testlib"
+
+test_setup() {
+
+	create_test_environment "$TEST_NAME"
+	create_bundle -n test-bundle1 -f /foo/test-file1 "$TEST_NAME"
+	create_bundle -n test-bundle2 -f /bar/test-file2 "$TEST_NAME"
+	create_bundle -n test-bundle3 -f /bar/test-file3 "$TEST_NAME"
+
+	# Create bundle dependencies
+	add_dependency_to_manifest "$WEBDIR"/10/Manifest.os-core test-bundle1
+	add_dependency_to_manifest -o "$WEBDIR"/10/Manifest.os-core test-bundle2
+	add_dependency_to_manifest -o "$WEBDIR"/10/Manifest.test-bundle1 test-bundle3
+
+}
+
+@test "DIA013: Diagnose a system that has a missing also-add" {
+
+	run sudo sh -c "$SWUPD diagnose $SWUPD_OPTS"
+
+	assert_status_is "$SWUPD_NO"
+	expected_output=$(cat <<-EOM
+		Diagnosing version 10
+		Checking for missing files
+		.* Missing file: .*/target-dir/foo
+		.* Missing file: .*/target-dir/foo/test-file1
+		.* Missing file: .*/target-dir/usr/share/clear/bundles/test-bundle1
+		Checking for corrupt files
+		Checking for extraneous files
+		Inspected 5 files
+		  3 files were missing
+		Use "swupd repair" to correct the problems in the system
+		Diagnose successful
+
+	EOM
+	)
+	assert_regex_is_output "$expected_output"
+
+}
```

Comment:
so if I understand this correctly, if an also-add bundle has never been removed explicitly by the user, a "swupd diagnose" should find it as "missing", right? Because also-add bundles should be added by default. Once a user removes it, a future "swupd diagnose" should not report it as missing, correct?

Reply:
There's no way for us in a diagnose to know if a bundle was removed by the user or if it wasn't added by a bundle-add.

If I run:

    $ swupd bundle-add --skip-optional bundle_name
    $ swupd diagnose

I would expect diagnose to be successful. Do this make sense?

Reply:
yes, my bad, for some reason I though test-bundle1 had been added as optional dependency to os-core, but I just realized it is an include, that is why it was added. 😊